File: app/src/main/java/com/attilapalfi/exceptional/services/Converter.java
package com.attilapalfi.exceptional.services;

import com.attilapalfi.exceptional.model.*;
import com.attilapalfi.exceptional.model.Exception;

import java.math.BigInteger;
import java.util.ArrayList;
import java.util.Collection;
import java.util.List;

/**
 * Created by palfi on 2015-08-19.
 */
public class Converter {

    public static List<BigInteger> fromFriendsToLongs(Collection<Friend> friends) {
        List<BigInteger> ids = new ArrayList<>(friends.size());
        for (Friend f : friends) {
            ids.add(f.getId());
        }
        return ids;
    }

    public static List<BigInteger> fromExceptionTypesToLongs(Collection<com.attilapalfi.exceptional.model.Exception> exceptions) {
        List<BigInteger> ids = new ArrayList<>(exceptions.size());
        for (Exception e : exceptions) {
            ids.add(e.getInstanceId());
        }
        return ids;
    }

}

File: app/src/main/java/com/attilapalfi/exceptional/ui/helpers/Converter.kt
package com.attilapalfi.exceptional.ui.helpers

import com.attilapalfi.exceptional.dependency_injection.Injector
import com.attilapalfi.exceptional.model.Exception
import com.attilapalfi.exceptional.model.Friend
import com.attilapalfi.exceptional.persistence.FriendStore
import java.math.BigInteger
import javax.inject.Inject

/**
 * Created by palfi on 2015-10-03.
 */
public class Converter {
    @Inject
    lateinit val friendStore: FriendStore

    init {
        Injector.INSTANCE.applicationComponent.inject(this)
    }

    public fun getNameAndCity(exception: Exception, fromWho: Friend): String {
        var senderName = fromWho.getName()
        return addCity(senderName, exception.city)
    }

    private fun getSenderName(exception: Exception): String {
        var senderName = ""
        if (exception.fromWho != BigInteger.ZERO ) {
            senderName = friendStore.findFriendById(exception.fromWho).getName()
        }
        return senderName
    }

    private fun addCity(senderName: String, city: String): String {
        var nameAndCity = senderName
        if ("" != city) {
            nameAndCity += (", " + city)
        }
        return nameAndCity
    }
}